src/e2e-app/BUILD.bazel

```diff
@@ -91,31 +105,57 @@ concatjs_devserver(
     additional_root_paths = [
         "npm/node_modules",
     ],
-    # Name of the AMD module that should be required on page load.
     entry_module = "angular_material/src/e2e-app/main",
     port = 4200,
-    # Scripts which will be included in the serving_path bundle after
-    # RequireJS has been loaded.
+    # Scripts which will be included in the serving_path bundle after RequireJS has been loaded.
     scripts = [
         ":devserver-configure.js",
         "//tools/rxjs:rxjs_umd_modules",
         "@npm//:node_modules/tslib/tslib.js",
     ] + getAngularUmdTargets(),
-    # Serving path of the bundle that serves all files specified in "deps" and "scripts".
+    # Name of the generated `concatjs` bundle that is built based on the labels in `deps` and `scripts`.
     serving_path = "/bundle.js",
-    # Files which should be provided by Bazel when running the devserver. These are not
-    # automatically served, but can be loaded manually through HTTP requests.
-    static_files = [
-        "@npm//zone.js",
-        "@npm//core-js-bundle",
-        "@npm//kagekiri",
-        "@npm//material-components-web",
-        "@npm//moment",
-        "@npm//@webcomponents/custom-elements",
-        "//src/material/prebuilt-themes:indigo-pink",
-        ":index.html",
-        ":theme",
-    ],
+    static_files = devserverIndexHtmlDependencies,
     tags = ["manual"],
     deps = [":e2e-app"],
 )
+
+# Bundles the `e2e-app` using rollup while running the Angular linker to transform
+# partial declarations to their corresponding Angular definitions. This target can
+# only be run if the e2e-app is built in partial compilation mode.
```

Comment:
> This target can only be run if the e2e-app is built in partial compilation mode.

Why? Running the linker on non-partial code is a noop.

Reply:
It definitely can; but shouldn't be. I want to throw an error if someone runs this linker-specific target without partial compilation. Updated the comment.